tests/datatypes/test_str.py

```diff
@@ -109,6 +109,42 @@ def test_slice(self):
             print(x[-1:0:-1])
             """)
 
+    def test_startswith(self):
+        self.assertCodeExecution("""
+            # Single character
+            if 'BeeWare'.startswith('B'):
+                print('BeeWare starts with B')
+            else:
+                print('BeeWare does not start with B')
+            # Substring
+            if 'BeeWare'.startswith('Bee'):
+                print('BeeWare starts with Bee')
+            else:
+                print('BeeWare does not start with Bee')
+            # Invalid substring
+            if 'BeeWare'.startswith('Ware'):
+                print('BeeWare starts with Ware')
+            else:
+                print('BeeWare does not start with Ware')
+            # Superstring
+            if 'BeeWare'.startswith('BeeWare-shaves-yaks'):
+                print('BeeWare starts with BeeWare-shaves-yaks')
+            else:
+                print('BeeWare does not start with BeeWare-shaves-yaks')
+            # Empty string
+            if 'BeeWare'.startswith(''):
+                print('BeeWare starts with an empty string')
+            else:
+                print('BeeWare does not start with an empty string')
+            # Invalid type
+            if 'BeeWare'.startswith(5):
```

Comment:
...and, this, boys and girls, is why we have CI. Of course - this one should raise an exception, not return False. So:

1) the patch needs to use a try-catch here to make sure the right exception is thrown; on throwing an exception, the test passes.

2) Either the test needs o be marked as an `@expectedFailure`, or the implementation needs a type check to throw the right exception.